File: Data Structures & Algorithms/house-robber/submission-1.cpp
class Solution
{
public:
    int rob(vector<int> &nums)
    {
        int size = nums.size();
        if (size == 1)
        {
            return nums[0];
        }
        int money[nums.size()];
        money[0] = nums[0];
        money[1] = max(nums[1], nums[0]);
        for (int i = 2; i < size; i++)
        {
            money[i] = max(money[i - 2] + nums[i], money[i - 1]);
        }

        return money[size - 1];
    }
};

File: Data Structures & Algorithms/missing-number/submission-1.py
from typing import List

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        max_num = max(nums)
        
        total_sum = sum(nums)
        
        new_num = (max_num) * (max_num + 1) // 2 - total_sum
        if new_num == 0 and 0 in nums:
            return max_num + 1
        return new_num